### PR TITLE
Response delay

### PR DESCRIPTION
**Pull request by wilversings**
Added response delay feature.  Needed this for one of my projects.

### CODE REVIEW

**Comment by hibri** [comment]
Hi @wilversings I'll look into this, sometime this week and merge it in. Apologies for the delay.

**Comment by StanislavRadkov** [comment]
Great work! Was looking for this functionality as well! 👍

File: src/HttpMock/IRequestStub.cs [hunk 1]
```diff
@@ -20,5 +20,7 @@ public interface IRequestStub
 		IRequestStub AddHeader( string header, string headerValue);
 		IRequestStub WithUrlConstraint(Func<string, bool> constraint);
 	    IRequestStub ReturnFileRange(string pathToFile, int from, int to);
+        IRequestStub AddDelay(uint milliseconds);
```
Comment (hibri) [inline src/HttpMock/IRequestStub.cs]: Rename to WithDelay

File: src/HttpMock/RequestProcessor.cs [hunk 2]
```diff
@@ -39,6 +40,8 @@ public class RequestProcessor :  IRequestProcessor
 	    private static void HandleRequest(HttpRequestHead request, IDataProducer body, IHttpResponseDelegate response, IRequestHandler handler)
 	    {
 	        _log.DebugFormat("Matched a handler {0}:{1} {2}", handler.Method, handler.Path, DumpQueryParams(handler.QueryParams));
+
+            Thread.Sleep(handler.ResponseDelay);
```
Comment (hibri) [inline src/HttpMock/RequestProcessor.cs]: Call Thread.Sleep only if there is a ResponsDelay > 0

**Comment by hibri** [comment]
@wilversings  Could you test for multiple stubs in the same test? 
If one stub has a response delay and another stub does not, this could block all responses, because of the Thread.Sleep

**Comment by wilversings** [comment]
@hibri Please check now

**Comment by hibri** [comment]
@wilversings thanks. Will take a look.

**Comment by hibri** [comment]
Thanks for this @wilversings . I'll make a small change to change the delay from uint to int before I publish the package

**Pull request merged**